src/lib/files.ts

```diff
@@ -0,0 +1,138 @@
+import { fabric } from "fabric";
+
+import { HistoryCommand } from "./history";
+import Pages, { PageJSON } from "./pages";
+
+export class AsyncReader extends Promise<FileReader> {
+  constructor(file: File) {
+    if (file instanceof File)
+      super((resolve, reject) => {
+        const reader = new FileReader();
+        reader.onload = () => {
+          resolve(reader);
+        };
+        reader.onerror = reject;
+        reader.readAsText(file);
+      });
+    else super(file);
+  }
+}
+
+// manages version compatibility with old document formats
+// change the signature and usages to accommodate new data; this will fill in sample data for missing fields
+export class JSONReader {
+  static async read(reader: AsyncReader): Promise<PageJSON[]> {
+    const json = (await reader).result.toString();
+    const object = JSON.parse(json);
+
+    // V0
+    if (Array.isArray(object)) return object;
```

Comment:
do we really need to check this

Reply:
istg no one uses this

Reply:
if you're talking about checking the type, this is how you ensure backwards compatibility when changing your file format.

though i do agree v0 (which doesn't specify qboard-version and is just a pure array) needs to be deprecated

Reply:
yeah let's just get rid of v0

Reply:
this breaks any files that were previously exported from your demo. is that fine?

Reply:
yes